onnxruntime/core/providers/cpu/tensor/gather_nd.cc

```diff
@@ -63,12 +71,20 @@ Status GatherNDBase::PrepareForCompute(OpKernelContext* context, Prepare& p) con
     element_counts[i] = input_shape.SizeFromDimension(i + 1);
   }
 
+  auto last_dim_size = indices_shape.SizeFromDimension(indices_shape.NumDimensions() - 1);
+#ifdef USE_OPENMP
+#pragma omp parallel for
+#endif
+  for (int64_t i = batch_dims_ - 1; i >= 0; --i) {
+    element_counts[last_indices_dimension + i] = indices_shape.SizeFromDimension(i + 1) / last_dim_size;
+  }
+
   int64_t err_index = 0;
   p.element_bytes = input_tensor->DataType()->Size();
   p.element_to_copy = input_shape.SizeFromDimension(last_indices_dimension);
   p.bytes_to_copy = p.element_bytes * p.element_to_copy;
-  const auto* indices_data = indices_tensor->Data<Tind>();
-  const int64_t offset_count = indices_shape.Size() / last_indices_dimension;  // Times to copy
+  const auto* indice_offset = indices_tensor->Data<Tind>();
+  const int64_t offset_count = indices_shape.Size() / (last_indices_dimension - batch_dims_);  // Times to copy
```

Comment:
I found the naming in this function is pretty confusing, (hard to understand), I can help rename them in my later PR.

Reply:
for example I thin calling slice_count is better than offset_count. 

element_counts[0:last_indices_dimension] is very different with element_counts[last_indices_dimension + batch_dims_ - 1:last_indices_dimension ]

Reply:
Yes. Please... 
Thanks for the help. 
I will leave the name untouched in this PR